### PR TITLE
Fix max series test loop that returns early

### PR DESCRIPTION
**Pull request by chroberts**
**What this PR does / why we need it**:

This PR fixes a test. Before the PR, the `TestEngine_MaxSeries` test `return`s after the first test and prevents the other tests from running.

**Which issue(s) this PR fixes**:
Fixes #: N/A

**Special notes for your reviewer**:
Adding `t.Run` allows using the `return`, but I went ahead and also removed the `return` and added an `else` to make it more clear. This would have prevented this and may protect against future moves/refactors.

**Checklist**
- [ ] Documentation added
- [x] Tests updated
- [ ] Is this an important fix or new feature? Add an entry in the `CHANGELOG.md`.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: pkg/logql/engine_test.go [hunk 1]
```diff
@@ -2168,21 +2168,23 @@ func TestEngine_MaxSeries(t *testing.T) {
...
+				require.True(t, errors.Is(err, logqlmodel.ErrLimit))
+			} else {
+				require.Nil(t, err)
+			}
 		})
-		_, err := q.Exec(user.InjectOrgID(context.Background(), "fake"))
-		if test.expectLimitErr {
-			require.NotNil(t, err)
-			require.True(t, errors.Is(err, logqlmodel.ErrLimit))
-			return
```
Comment (DylanGuedes) [inline pkg/logql/engine_test.go:2183]: hmm if you modify the return to be a `continue` the wrapping inside the t.Run() isn't necessary I think. Do you mind double checking it?

**Review by DylanGuedes — CHANGES_REQUESTED** [review]

File: pkg/logql/engine_test.go (hunk 1 above)
Comment (chroberts) [reply pkg/logql/engine_test.go:2183]: Yes, that is correct. Isn't `t.Run` preferred when making derived tests?

Comment (DylanGuedes) [reply pkg/logql/engine_test.go:2183]: it is, you're right. I thought that you wrapped it as a solution to the early return but you actually did it to improve things.

**Review by DylanGuedes — APPROVED** [review]
lgtm

File: pkg/logql/engine_test.go (hunk 1 above)
Comment (cstyan) [reply pkg/logql/engine_test.go:2183]: > Isn't t.Run preferred when making derived tests?

generally yes, it makes running one of the specific sub tests possible

**Review by cyriltovena — APPROVED** [review]
LGTM

**Pull request merged**